File: 1127-last-stone-weight/1127-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> queue = new PriorityQueue<>((a, b) -> b - a);

        for (int stone : stones) {
            queue.add(stone);
        }

        while (queue.size() >= 2) {
            int s1 = queue.poll();
            int s2 = queue.poll();
            if (s1 != s2) {
                queue.add(Math.abs(s1 - s2));
            }
        }

        if (queue.isEmpty()) {
            return 0;
        } else {
            return queue.poll();
        }
    }
}

File: 1127-last-stone-weight/1127-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        queue = []

        for stone in stones:
            heapq.heappush(queue, -stone)
        
        while len(queue) > 1:
            s1 = -heapq.heappop(queue)
            s2 = -heapq.heappop(queue)

            if s1 != s2:
                heapq.heappush(queue, -abs(s1 - s2))
        
        return -queue[0] if len(queue) == 1 else 0